airflow/models/dag.py

```diff
@@ -545,7 +552,10 @@ def __init__(
         self.template_undefined = template_undefined
         self.last_loaded = timezone.utcnow()
         self.safe_dag_id = dag_id.replace(".", "__dot__")
-        self.max_active_runs = max_active_runs
+        if isinstance(self.timetable, ContinuousTimetable):
```

Comment:
It feels kind of messy to have this sort of conditional override in the `__intit__`, but I'm not sure where to put this important logic 🤔 

Thoughts?

Reply:
We could add a new hook in the timetable protocol for this. Most timetables would return None (not limit) and ContinuousTimetable returns 1.

Reply:
This is a good idea, and could probably be used for other future timetables - thanks for the suggestion, I'll take a look at this later today / maybe tomorrow.